### PR TITLE
Update documentation for torch.mean() to note behavior with empty tensors

### LINKED ISSUE #141057: Ambiguous Behavior When Computing Mean of Empty Tensor Returns
### 🐛 Describe the bug

When calculating the mean of an empty tensor using functions like torch.mean(), the result is nan (Not a Number). While this is mathematically consistent, it can lead to unexpected behavior in models, especially when computing loss functions where an empty batch might occur due to data filtering or dynamic batching. This behavior is not explicitly documented, and handling nan values can complicate training loops.

During unit testing of a custom loss function, it was observed that computing the mean over empty tensors resulted in nan, which subsequently affected model training by introducing nan values into gradients and parameters.

### Versions

Collecting environment information...
PyTorch version: 2.5.0+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 Home (10.0.22631 64-bit)
GCC version: (MinGW.org GCC-6.3.0-1) 6.3.0
Clang version: Could not collect
CMake version: version 3.30.5
Libc version: N/A

Python version: 3.11.5 | packaged by Anaconda, Inc. | (main, Sep 11 2023, 13:26:23) [MSC v.1916 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.22631-SP0
Is CUDA available: True
CUDA runtime version: 12.5.40
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 4080
Nvidia driver version: 561.09
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Name: Intel(R) Core(TM) i7-14700K
Manufacturer: GenuineIntel
Family: 198
Architecture: 9
ProcessorType: 3
DeviceID: CPU0
CurrentClockSpeed: 3400
MaxClockSpeed: 3400
L2CacheSize: 28672
L2CacheSpeed: None
Revision: None

Versions of relevant libraries:
[pip3] flake8==6.0.0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.24.3
[pip3] numpydoc==1.5.0
[pip3] torch==2.5.0+cu124
[pip3] torchaudio==2.5.0+cu124
[pip3] torchvision==0.20.0
[conda] _anaconda_depends         2023.09             py311_mkl_1
[conda] blas                      1.0                         mkl
[conda] mkl                       2023.1.0         h6b88ed4_46357
[conda] mkl-service               2.4.0           py311h2bbff1b_1
[conda] mkl_fft                   1.3.8           py311h2bbff1b_0
[conda] mkl_random                1.2.4           py311h59b6b97_0
[conda] numpy                     1.24.3          py311hdab7c0b_1
[conda] numpy-base                1.24.3          py311hd01c5d8_1
[conda] numpydoc                  1.5.0           py311haa95532_0
[conda] torch                     2.5.0+cu124              pypi_0    pypi
[conda] torchaudio                2.5.0+cu124              pypi_0    pypi
[conda] torchvision               0.20.0                   pypi_0    pypi

### PR DESCRIPTION
**Pull request by PadenMilligan**
This PR updates the documentation for `torch.mean()` to explicitly mention that computing the mean over an empty tensor returns `nan`. This clarification helps users understand the behavior and handle it appropriately in their code.

Fixes #141057

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Thanks
